src/compiler/checker.rs

```diff
@@ -2448,20 +2587,25 @@ fn create_node_builder() -> NodeBuilder {
     NodeBuilder::new()
 }
 
-pub struct NodeBuilder {}
+pub struct NodeBuilder {
+    synthetic_factory: BaseNodeFactorySynthetic,
+}
 
 impl NodeBuilder {
     pub fn new() -> Self {
-        Self {}
+        Self {
+            synthetic_factory: get_synthetic_factory(),
```

Comment:
Per a previous PR comment because of not being able to use `lazy_static!` for the `synthetic_factory` that the default singleton node `factory` wants to use as its `BaseNodeFactory` I'm instead calling `get_synthetic_factory()` to be able to inject that synthetic factory into calls into `factory` methods. So for `NodeBuilder` which is frequently calling into `factory` I figured I can give it a single synthetic factory instance upon creation and have it inject that rather than have the boilerplate + cost of calling `get_synthetic_factory()` inside any individual `NodeBuilder` method that needs to call into `factory`